jobrunner/job_executor.py

```diff
@@ -0,0 +1,135 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Protocol, Mapping, List, Tuple, Optional
+
+from jobrunner.models import State, StatusCode
+
+
+class Privacy(Enum):
+    HIGH = "high"
+    MEDIUM = "medium"
+
+
+@dataclass
+class Study:
+    git_repo_url: str
+    commit: str
+
+
+@dataclass
+class JobDefinition:
+    id: str  # a unique identifier for the job
+    study: Study  # the study defining the action for this job
+    workspace: str  # the workspace to run the job in
+    action: str  # the name of the action that the job is running
+    image: str  # the Docker image to run
+    args: List[str]  # the arguments to pass to the Docker container
+    env: Mapping[str, str]  # the environment variables to set for the Docker container
+    inputs: List[str]  # the files that the job requires
+    output_spec: Mapping[str, str]  # the files that the job should produce (globs mapped to privacy levels)
```

Comment:
Drive-by thought (not sure of the best place to capture it). Can we/should we specify that output patterns are supplied as some dialect of regular expression (e.g POSIX or POSIX-extended) rather than "globs"? That gives us a reasonable amount of flexibility in our matching logic, and it avoids the problem that the interpretation of glob-like syntax is under-specified (e.g do wildcards match slashes or not?)